### PR TITLE
chore: download asset bundles for local scenes

### PR DESCRIPTION
**Pull request by dalkia**
## What does this PR change?

ONLY FOR DEBUGGING

Creates a HybridSceneApproach (LocalScene + Remote Asset bundle)
If the dev inputs the remote scene id and the content server correctly, they should be able to debug a scene with remote asset bundles. 
If the remote assets do not exist, regular pipeline will continue

## How to test the changes?

1. Launch the explorer and do the regular stuff. The common user should see no changes

## Our Code Review Standards

https://github.com/decentraland/unity-renderer/blob/master/docs/code-review-standards.md

### CODE REVIEW

**Review by sandrade-dcl — APPROVED** [review]

File: Explorer/Assets/Scripts/Global/Dynamic/MainSceneLoader.cs [hunk 1]
```diff
@@ -175,6 +179,14 @@ private async UniTask InitializeFlowAsync(CancellationToken ct)
 
                 bool shouldEnableLandscape = enableLandscape;
 
+                var hybridSceneParams = new HybridSceneParams();
+                if (useRemoteAssetBundles && initialRealm == InitialRealm.Localhost)
+                {
+                    hybridSceneParams.EnableHybridScene = true;
```
Comment (mikhail-dcl) [inline Explorer/Assets/Scripts/Global/Dynamic/MainSceneLoader.cs:185]: I think we can remove this flag and assume it's always true. Is there a scenario when you don't want to enable it?

File: Explorer/Assets/Scripts/ECS/SceneLifeCycle/SceneFacade/Systems/LoadHibridSceneSystemLogic.cs [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+using ECS.SceneLifeCycle.Components;
+using ECS.SceneLifeCycle.SceneDefinition;
+using SceneRunner;
+using SceneRunner.Scene;
+using UnityEngine;
+using Utility;
+
+namespace ECS.SceneLifeCycle.Systems
+{
+    public class LoadHibridSceneSystemLogic : LoadSceneSystemLogicBase
```
Comment (mikhail-dcl) [inline Explorer/Assets/Scripts/ECS/SceneLifeCycle/SceneFacade/Systems/LoadHibridSceneSystemLogic.cs:18]: Please correct a typo everywhere, it's "Hybrid"

File: Explorer/Assets/Scripts/ECS/SceneLifeCycle/SceneFacade/Systems/LoadHibridSceneSystemLogic.cs [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+            var definition = definitionComponent.Definition;
+
+            // Warning! Obscure Logic!
+            // Each scene can override the content base url, so we need to check if the scene definition has a base url
+            // and if it does, we use it, otherwise we use the realm's base url
+            var contentBaseUrl = ipfsPath.BaseUrl.IsEmpty
+                ? intention.IpfsRealm.ContentBaseUrl
+                : ipfsPath.BaseUrl;
+
+            ISceneContent hashedContent = new SceneHashedContent(definition.content, contentBaseUrl);
```
Comment (mikhail-dcl) [inline Explorer/Assets/Scripts/ECS/SceneLifeCycle/SceneFacade/Systems/LoadHibridSceneSystemLogic.cs:42]: The only difference here is how you create `ISceneContent`, everything else is copied, thus I propose to create a single abstract method to provide `ISceneContent` and everything else can be in the base class

**Review by mikhail-dcl — CHANGES_REQUESTED** [review]

File: Explorer/Assets/Scripts/ECS/SceneLifeCycle/SceneFacade/Systems/LoadHibridSceneSystemLogic.cs (hunk 2 above)
Comment (dalkia) [reply Explorer/Assets/Scripts/ECS/SceneLifeCycle/SceneFacade/Systems/LoadHibridSceneSystemLogic.cs:18]: Damn spanish xD. Correcting!

File: Explorer/Assets/Scripts/Global/Dynamic/MainSceneLoader.cs (hunk 1 above)
Comment (dalkia) [reply Explorer/Assets/Scripts/Global/Dynamic/MainSceneLoader.cs:185]: Agreed. Removing it from the if and setting it always true

File: Explorer/Assets/Scripts/ECS/SceneLifeCycle/SceneFacade/Systems/LoadHibridSceneSystemLogic.cs (hunk 3 above)
Comment (dalkia) [reply Explorer/Assets/Scripts/ECS/SceneLifeCycle/SceneFacade/Systems/LoadHibridSceneSystemLogic.cs:42]: Done. The methods in `LoadSceneSystemLogic` feel a little bit redudant, but needed to use the abstraction

**Pull request merged**